packages/react-native-audio-api/common/cpp/audioapi/core/sources/StreamerNode.cpp

```diff
@@ -0,0 +1,314 @@
+#include <audioapi/core/BaseAudioContext.h>
+#include <audioapi/core/sources/StreamerNode.h>
+#include <audioapi/core/utils/Locker.h>
+#include <audioapi/dsp/VectorMath.h>
+#include <audioapi/utils/AudioArray.h>
+#include <audioapi/utils/AudioBus.h>
+#include <chrono>
+
+namespace audioapi {
+StreamerNode::StreamerNode(BaseAudioContext *context)
+    : AudioScheduledSourceNode(context),
+      fmtCtx_(nullptr),
+      codecCtx_(nullptr),
+      decoder_(nullptr),
+      codecpar_(nullptr),
+      pkt_(nullptr),
+      frame_(nullptr),
+      pendingFrame_(nullptr),
+      bufferedBus_(nullptr),
+      bufferedBusIndex_(0),
+      maxBufferSize_(0),
+      audio_stream_index_(-1),
+      swrCtx_(nullptr),
+      resampledData_(nullptr),
+      maxResampledSamples_(0) {}
+
+StreamerNode::~StreamerNode() {
+  cleanup();
+}
+
+bool StreamerNode::initialize(const std::string &input_url) {
+  if (isInitialized_) {
+    cleanup();
+  }
+
+  if (!openInput(input_url)) {
+    return false;
+  }
+
+  if (!findAudioStream()) {
```

Comment:
Why not squash the three next ifs into one if? Less verbose and equally readable imo